flows/contact.go

```diff
@@ -360,23 +382,33 @@ func (c *Contact) ReevaluateDynamicGroups(env envs.Environment) ([]*Group, []*Gr
 	removed := make([]*Group, 0)
 	errs := make([]error, 0)
 
-	for _, group := range c.assets.Groups().All() {
-		if !group.IsDynamic() {
-			continue
+	if c.Blocked() || c.Stopped() {
+		for _, group := range c.assets.Groups().All() {
+			if c.groups.Remove(group) {
+				removed = append(removed, group)
+			}
 		}
 
-		qualifies, err := group.CheckDynamicMembership(env, c)
-		if err != nil {
-			errs = append(errs, errors.Wrapf(err, "unable to re-evaluate membership of group '%s'", group.Name()))
-		}
+	} else {
 
-		if qualifies {
-			if c.groups.Add(group) {
-				added = append(added, group)
+		for _, group := range c.assets.Groups().All() {
+			if !group.IsDynamic() {
+				continue
 			}
-		} else {
-			if c.groups.Remove(group) {
-				removed = append(removed, group)
+
+			qualifies, err := group.CheckDynamicMembership(env, c)
```

Comment:
Ok sorry one more tweak!  Just push the check for stopped/blocked into `Group.CheckDynamicMembership`.. then you don't need any extra logic here. All the groups will return false. 

Plus it's good to have that method be correct for stopped/blocked contacts.

We're essentially saying every dynamic group has an implicit `AND stopped = false AND blocked = false in its query`

Reply:
What about the static groups? I think the logic we want is to remove them as well

Reply:
Ah good point. I still think we want `Group.CheckDynamicMembership` and `Contact.ReevaluateDynamicGroups` to be correct for stopped/blocked.

How about rename `BaseModifier.reevaluateDynamicGroups` to `BaseModifier.reevaluateGroups` and give it a new bool arg like `removeStatic` that you set to true if `stopped||blocked`

Then it can include the static groups in the event that it generates?

Reply:
I updated that